protobuf/protoc-gen-govalidator/plugin/plugin.go

```diff
@@ -913,13 +913,16 @@ func genMsgValidator(g *protogen.GeneratedFile, f *protogen.Field, varName strin
 	if repeated {
 		// Please leave the len check in place here as I have identified edge cases where it is required (will add test case for it later)
 		g.P("if len(", varName, ") > 0 {")
-		g.P("for _, item := range ", varName, "{")
+		g.P("for i, item := range ", varName, "{")
 		varName = "item"
 	}
 
 	g.P("if ", varName, " != nil {")
 	g.P("if v, ok := interface{}(", varName, ").(", s12protoPackage.Ident("Validator"), "); ok {")
 	g.P("if err := v.Validate(); err != nil {")
+	if repeated {
+		g.P("err = ", fmtPackage.Ident("Errorf"), "(\"at index %d: %v\", i, err)")
```

Comment:
Thanks for adding this @lachlancooper. Did you check how this works with multiple validation errors in different messages? We should also add tests for this to make sure that it's working as expected. I can take a look later in the week if you'd like.

Reply:
Good question, I guess currently validation errors don't report the specific message that failed?

I agree about adding tests, will move this to draft until we get a chance to address that.